lib/main.js

```diff
@@ -245,35 +249,61 @@ var reporterPanel = panel.Panel({
     }
 });
 
+function retrieveAddonIcon(addon, callback) {
+    if (!addon.iconURL) {
+        addon.icon = genericAddonIconURL;
+        callback(addon);
+    } else {
+        var xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function () {
+            if (this.readyState === 4) {
+            addon.icon = 'data:image/png;base64,' + base64.encode(String.fromCharCode.apply(null, new Uint8Array(this.response)));
```

Comment:
You should test this for large images. I believe in some environments there is a limit as to how many arguments can be passed. Maybe use `(new UInt8Array(this.response)).map(c => String.fromCharCode(c)).join('')` or use `.subarray()` to split it into chunks.